From c41ca5d886f614484baf1e8f17d79a433f485c8b Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Thu, 28 Apr 2016 21:15:38 +0800
Subject: [PATCH 1193/5242] MLK-12928-10 mfd: pf1550: correct virtual irqs for
 every sub-driver of pf1550

commit  ba5cd77c08ae2b5a5da147dc7bf7e57cc9ed796c from
https://source.codeaurora.org/external/imx/linux-imx.git

Correct virtual irqs macro from zero for every sub-driver of pf1550,
otherwise,below warning will be triggered:
Signed-off-by: Meng Li <Meng.Li@windriver.com>
------------[ cut here ]------------
WARNING: CPU: 0 PID: 1 at kernel/irq/irqdomain.c:280 irq_domain_associate+0x148/0x1d4()
error: hwirq 0xb is too large for (null)
Modules linked in:
CPU: 0 PID: 1 Comm: swapper/0 Not tainted 4.1.15-01689-gb67ecb6-dirty #195
Hardware name: Freescale i.MX6 Ultralite (Device Tree)
[<80015e04>] (unwind_backtrace) from [<80012754>] (show_stack+0x10/0x14)
[<80012754>] (show_stack) from [<807909c0>] (dump_stack+0x84/0xc4)
[<807909c0>] (dump_stack) from [<80034914>] (warn_slowpath_common+0x84/0xb4)
[<80034914>] (warn_slowpath_common) from [<80034974>] (warn_slowpath_fmt+0x30/0x40)
[<80034974>] (warn_slowpath_fmt) from [<800717fc>] (irq_domain_associate+0x148/0x1d4)
[<800717fc>] (irq_domain_associate) from [<80071da4>] (irq_create_mapping+0x60/0xc4)
[<80071da4>] (irq_create_mapping) from [<804910a4>] (pf1550_onkey_probe+0xe8/0x230)
[<804910a4>] (pf1550_onkey_probe) from [<803823e8>] (platform_drv_probe+0x44/0xa4)
[<803823e8>] (platform_drv_probe) from [<80380ca0>] (driver_probe_device+0x174/0x2b4)
[<80380ca0>] (driver_probe_device) from [<80380eb0>] (__driver_attach+0x8c/0x90)
[<80380eb0>] (__driver_attach) from [<8037f1e8>] (bus_for_each_dev+0x6c/0xa0)
[<8037f1e8>] (bus_for_each_dev) from [<8038043c>] (bus_add_driver+0x148/0x1f0)
[<8038043c>] (bus_add_driver) from [<803814b4>] (driver_register+0x78/0xf8)
[<803814b4>] (driver_register) from [<80009730>] (do_one_initcall+0x8c/0x1d4)
[<80009730>] (do_one_initcall) from [<80a8bdac>] (kernel_init_freeable+0x144/0x1e4)
[<80a8bdac>] (kernel_init_freeable) from [<8078ca20>] (kernel_init+0x8/0xe8)
[<8078ca20>] (kernel_init) from [<8000f568>] (ret_from_fork+0x14/0x2c)
---[ end trace dc402f301115a3b2 ]---

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
---
 include/linux/mfd/pf1550.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/linux/mfd/pf1550.h b/include/linux/mfd/pf1550.h
index 8066b8c..f19c493 100644
--- a/include/linux/mfd/pf1550.h
+++ b/include/linux/mfd/pf1550.h
@@ -182,7 +182,7 @@ enum pf1550_pmic_reg {
 #define CHARG_IRQ_DPMI		BIT(6)
 #define CHARG_IRQ_THMI		BIT(7)
 
-enum pf1550_irq {
+enum pf1550_pmic_irq {
 	PF1550_PMIC_IRQ_SW1_LS,
 	PF1550_PMIC_IRQ_SW2_LS,
 	PF1550_PMIC_IRQ_SW3_LS,
@@ -194,14 +194,18 @@ enum pf1550_irq {
 	PF1550_PMIC_IRQ_LDO3_FAULT,
 	PF1550_PMIC_IRQ_TEMP_110,
 	PF1550_PMIC_IRQ_TEMP_125,
+};
 
+enum pf1550_onkey_irq {
 	PF1550_ONKEY_IRQ_PUSHI,
 	PF1550_ONKEY_IRQ_1SI,
 	PF1550_ONKEY_IRQ_2SI,
 	PF1550_ONKEY_IRQ_3SI,
 	PF1550_ONKEY_IRQ_4SI,
 	PF1550_ONKEY_IRQ_8SI,
+};
 
+enum pf1550_charg_irq {
 	PF1550_CHARG_IRQ_BAT2SOCI,
 	PF1550_CHARG_IRQ_BATI,
 	PF1550_CHARG_IRQ_CHGI,
-- 
1.7.9.5

